app/grandchallenge/discussion_forums/views.py

```diff
@@ -124,10 +125,24 @@ def get_context_data(self, **kwargs):
                 "post_create_form": ForumPostForm(
                     user=self.request.user, topic=self.topic
                 ),
+                "unread_posts_by_user": self.topic.get_unread_topic_posts_for_user(
+                    user=self.request.user
+                ),
             }
         )
         return context
 
+    def get(self, request, *args, **kwargs):
+        response = super().get(request, *args, **kwargs)
+
+        # mark topic as read by the user
+        TopicReadRecord.objects.update_or_create(
+            user=self.request.user,
```

Comment:
Shouldn't the anonymous user be excluded?

Reply:
The anonymous user does not have permission to access this view, and so `self.request.user` would never be the anonymous user here. I can add an explicit check, of course, but I think it's not strictly necessary.

Reply:
I have added the extra check just in case